java-checks-test-sources/src/main/java/checks/DateTimeFormatterMismatch.java

```diff
@@ -0,0 +1,80 @@
+package checks;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+import java.time.temporal.WeekFields;
+import java.util.Locale;
+
+public class DateTimeFormatterMismatch {
+  public void createUsingPatterns() {
+    DateTimeFormatter.ofPattern("Y-ww"); // Compliant
+    DateTimeFormatter.ofPattern("YY-ww"); // Compliant
```

Comment:
For completeness, we could add 2 tests for `ofPattern`: one where the argument is not a string literal, and another where the pattern does not contain any of the character we're looking for